pkg/pillar/cmd/baseosmgr/handlebaseos.go

```diff
@@ -846,7 +849,37 @@ func validateBaseOsConfig(ctx *baseOsMgrContext, config types.BaseOsConfig) erro
 	return nil
 }
 
-func handleZbootTestComplete(ctx *baseOsMgrContext, config types.ZbootConfig, status types.ZbootStatus) {
+func completeUpgrade(version string) {
+	lastImageVersion := types.LastImageVersion{}
+	lastImageVersion.Version = version
+	lastImageVersion.LastUpgradeTime = time.Now()
```

Comment:
This should be owned by the entity which does the upgrade, which currently is device-steps.sh

Reply:
ddevice-steps.sh starts the upgrade.. And baseosMgr is the one which completes the upgrade - isn't it? That that is the entity that knows upgade has completed. Hence it is generated by baseos mgr.